### PR TITLE
[docs] Generate navigation data with babel preval

### PR DESCRIPTION
**Pull request by nicknovitski**
# Why

I have hopes of eliminating the mdjs scripts entirely and writing [mdx](https://mdxjs.com/) files directly.  This brings us one step closer.

Also, more immediately, it enables the dynamic rearranging of sidebar links in development.

### CODE REVIEW

File: docs/common/navigation-data.js [hunk 1]
```diff
@@ -1,16 +1,8 @@
...
-const {
-  ROOT,
-  INTRODUCTION,
-  WORKING_WITH_EXPO,
-  GUIDES,
-  DISTRIBUTION,
-  EXPOKIT,
-  REACT_NATIVE,
-} = require('./sidebar-navigation-order');
+const fm = require('front-matter');
```
Comment (brentvatne) [inline docs/common/navigation-data.js]: unrelated to this pr but this looks unused

**Comment by brentvatne** [comment]
also i didn't know about preval before, it makes a lot of sense here 👌

**Comment by nicknovitski** [comment]
The failure is because links to `unversioned` are showing up in the production export somewhere.  I probably need to filter that out of the navigation links.

**Comment by nicknovitski** [comment]
Fixed, and ran prettier on changed files.

File: docs/common/navigation-data.js [hunk 2]
```diff
@@ -123,6 +72,15 @@ const generateNavLinks = (path_, arr) => {
...
+const ORIGINAL_PATH_PREFIX = './versions';
+
+let versionDirectories = fs
+  .readdirSync(ORIGINAL_PATH_PREFIX, { withFileTypes: true })
+  .filter(f => f.isDirectory())
+  .map(f => f.name);
+
+module.exports = versionDirectories.reduce(
+  (obj, version) =>
+    Object.assign({ [version]: generateNavLinks(`${ORIGINAL_PATH_PREFIX}/${version}`, []) }, obj),
```
Comment (brentvatne) [inline docs/common/navigation-data.js]: `{ ...obj, [version]: generateNavEtc }` seems a bit more clean to me but don't mind so much either way

**Review by brentvatne — APPROVED** [review]

File: docs/common/navigation-data.js (hunk 2 above)
Comment (nicknovitski) [reply docs/common/navigation-data.js]: Oh, good idea.

**Pull request merged**